static/app/test/specs/controller.spec.js

```diff
@@ -78,42 +82,69 @@ describe('Controller', function(){
     });
 
     describe('initialize', function(){
-        it('registers call:method request handler');
+        it('registers call:method request handler', function(){
+            var args = 'something';
+            var editStub = sinon.stub(this.controller, 'edit');
+            channel.trigger('call:method', 'edit', args);
+            assert.isTrue(editStub.calledOnce, 'editStub is called');
+            assert.isTrue(editStub.calledWith(args), 'editStub is passed an argument');
+        });
     });
 
     describe('destroy', function(){
-        it('unregisters call:method request handler');
+        it('unregisters call:method request handler', function(){
+            var editStub = sinon.stub(this.controller, 'edit');
+            this.controller.destroy();
+            channel.trigger('call:method', 'edit');
+            assert.isFalse(editStub.calledOnce, 'call:method does not call method');
+        });
     });
 
     describe('list', function(){
         it('shows loading view before view is loaded', function(){
-            var controller = new Controller();
             assert.isFalse(this.mainRegion.hasView(), 'no view shown yet');
-            controller.list();
+            this.controller.list();
             assert.isTrue(this.mainRegion.hasView(), 'a view in the mainRegion');
             assert.isTrue(this.mainRegion.currentView instanceof LoadingView, 'view in mainRegion is LoadingView');
         });
 
         it('shows list LayoutView after moves and spots are fetched', function(){
-            var controller = new Controller();
-            controller.list();
+            this.controller.list();
             this.server.respond();
             assert.isTrue(this.mainRegion.hasView(), 'a view in the mainRegion');
             assert.isTrue(this.mainRegion.currentView instanceof LayoutView, 'view in mainRegion is LayoutView');
         });
 
         it('passes saveAlert to LayoutView as recentSave option', function(){
-            var controller = new Controller();
-            controller.list(true);
+            this.controller.list(true);
             this.server.respond();
             assert.isTrue(this.mainRegion.currentView.getOption('recentSave'), 'passes true if save alert');
         });
     });
 
     describe('edit', function(){
-        it('inserts view immediately if spots have already been fetched');
-        it('inserts loading view first if spots have not already been fetched');
-        it('inserts NameView if move does not a have user name');
-        it('inserts MoveFormView if move does have a name');
+        it('inserts view immediately if spots have already been fetched', function(){
+            this.spots.fetch().done(function(){
+                this.controller.edit();
+                assert.isTrue(this.mainRegion.currentView instanceof NameView, 'view in mainRegion is NameView');
```

Comment:
This assertion isn't actually running. For our frontend tests, we use the sinon [fake server](http://sinonjs.org/docs/#fakeServer) to handle responding to AJAX requests because we aren't actually running a server. In this test suite, the fake server is set up [in the beforeEach block](https://github.com/Nestio/lunchmove/blob/tests_controller/static/app/test/specs/controller.spec.js#L61). The sinon fake server API requires us to actually call `server.respond` for it to respond. So what's happening here is that you're calling fetch and registering the `done` callback, but the callback never gets called. The reason why the test is 'passing' is because a test with no assertions automatically counts as passing. To have the test work, you want to do something like this:
```javascript
this.spots.fetch();
this.server.respond();
this.controller.edit();
assert.isTrue(this.mainRegion.currentView instanceof NameView, 'view in mainRegion is NameView');
```